src/util/sample.cpp

```diff
@@ -16,42 +17,55 @@ typedef qint32 int32_t;
 // This also utilizes AVX registers when compiled for a recent 64-bit CPU
 // using scons optimize=native.
 
+namespace {
+
+#ifdef __AVX__
+constexpr size_t kAlignment = 32;
+#else
+constexpr size_t kAlignment = 16;
+#endif
+
 // TODO() Check if uintptr_t is available on all our build targets and use that
 // instead of size_t, we can remove the sizeof(size_t) check than
-static inline bool useAlignedAlloc() {
+constexpr bool useAlignedAlloc() {
     // This will work on all targets and compilers.
     // It will return true on MSVC 32 bit builds and false for
     // Linux 32 and 64 bit builds
```

Comment:
isn't it an issue that it's returning false for Linux 32-bit builds? the failing test is on Ubuntu 32-bit?

Reply:
Nevermind! I didn't know about max_align_t

Reply:
Right, the comment is wrong, now that we know better. I will fix it tonight.